pandas/tests/indexes/test_indexing.py

```diff
@@ -96,3 +99,31 @@ def test_getitem_deprecated_float(idx):
 
     expected = idx[1]
     assert result == expected
+
+
+@pytest.mark.parametrize(
+    "idx, target, expected",
+    [
+        ([np.nan, "var1", np.nan], [np.nan], np.array([0, 2], dtype=np.int64)),
```

Comment:
Can you add a test case that includes both np.nan and 0 to make sure this can disambiguate?

Reply:
Good point, I don't think this will differentiate between the two because the conversion from np.nan to 0 happens before the check. I have a few of ideas:
1) Should we choose another replace value for np.nan with (ie. 'NaN')?
2) Check to see if targets/values have 0s in it and replace it with something else?
Let me know what direction you think I should go? Thanks!